### DISCUSSION TITLE
Various Issues with LDAP Source after 2023.6.0 upgrade

### DISCUSSION THREAD

**Discussion by AndrewBucklin** [goauthentik/authentik#6185] [Q&A] [answered]
After upgrading to 2023.6.0, the LDAP sync appears to be syncing 5 users at a time and creating a ton of entries in the Sync Status window. Is this normal? This list continues to grow, but here's what it looks like right now:

> Sync status
> 
> ldap_sync:default:users:72ee93db-39d0-406e-8058-b814a0930744
> 
> Last sync: 7/7/2023, 4:48:41 PM
> Synced 5 objects.
> ldap_sync:default:users:1e498568-f2c3-4212-b090-dbc1fcde940a
> 
> Last sync: 7/7/2023, 4:48:12 PM
> Synced 5 objects.
> ldap_sync:default:users:e62eb0d7-d3c3-46b8-8a74-36544951df0c
> 
> Last sync: 7/7/2023, 4:48:52 PM
> Synced 5 objects.
> ldap_sync:default:membership
> 
> Last sync: 7/7/2023, 4:34:08 PM
> Synced 5736 objects.
> ldap_sync:default:users:5786c286-be82-4e13-a206-fd38b792d59c
> 
> Last sync: 7/7/2023, 4:50:11 PM
> Synced 5 objects.
> ldap_sync:default:users:8c4601fa-c3a4-4fbd-a49c-e74bf7f9a280
> 
> Last sync: 7/7/2023, 4:50:22 PM
> Synced 5 objects.
> ldap_sync:default:users:1f9f4657-0053-4bb7-82f0-f89f163c2ec3
> 
> Last sync: 7/7/2023, 4:47:36 PM
> Synced 5 objects.
> ldap_sync:default:users:0411e4c4-b427-413e-9870-469643827f87
> 
> Last sync: 7/7/2023, 4:50:52 PM
> Synced 5 objects.
> ldap_sync:default:users:0561cadd-2f8e-4432-befb-fe100ff779bf
> 
> Last sync: 7/7/2023, 4:51:02 PM
> Synced 5 objects.
> ldap_sync:default:users:a0965cd3-306f-48d2-887b-88a0559fe627
> 
> Last sync: 7/7/2023, 4:48:50 PM
> Synced 5 objects.
> ldap_sync:default:users:27357591-647d-4765-9256-127c5b03a11a
> 
> Last sync: 7/7/2023, 4:49:56 PM
> Synced 5 objects.
> ldap_sync:default:users:2e7429e0-3b5f-4fa7-8df9-e270c85697ec
> 
> Last sync: 7/7/2023, 4:50:17 PM
> Synced 5 objects.
> ldap_sync:default:group
> 
> Last sync: 7/7/2023, 4:34:03 PM
> Synced 98 objects.
> ldap_sync:default:users:7bbad8c6-1612-460c-ae2e-ff93cda66de9
> 
> Last sync: 7/7/2023, 4:49:36 PM
> Synced 5 objects.
> ldap_sync:default:users:8e981fc1-1976-411b-93c0-5d005300ab74
> 
> Last sync: 7/7/2023, 4:49:31 PM
> Synced 5 objects.
> ldap_sync:default:user
> 
> Last sync: 7/7/2023, 2:56:18 PM
> Synced 156 objects.

**Comment by BeryJu** [comment]
Oh that's embarrassing, during testing the page size was set to 5 objects and I didn't change it after testing. See #6187, which will make it default to 50 objects per page and also make it adjustable

**Comment by AndrewBucklin** [comment]
Thanks. For now, I’ve disabled LDAP sync, to stop it from making a ton of additional entries in the Sync Status screen. 

I assume the fix will be in 2023.6.1? I’m
sure it’s too soon to tell, but any idea when that will go out?

Also, will the issue self-resolve or is there any cleanup that needs to be done to delete the slew of additional items in the Sync Status section?

**Reply by BeryJu** [reply]
You can continue to let the sync run, and the issue will fix itself, as the sync status is stored in redis and will expire after some time

2023.6.1 will be out relatively soon since there seem to be some other issues with the LDAP outpost

**Comment by AndrewBucklin** [comment]
I changed it from 5 to 50 and then re-enabled LDAP sync. Now I'm getting `'UUID' object has no attribute 'connection'` error. Any ideas?
[Error log](https://pastebin.com/EHGFRtfQ)

**Reply by BeryJu** [reply]
https://github.com/goauthentik/authentik/pull/6191

**Comment by AndrewBucklin** [comment]
Group and Membership are not syncing. Here's the relevant error: [Error log](https://pastebin.com/KTD8Z9Np)

**Reply by AndrewBucklin** [reply]
2023.6.1 seems to have resolved this error but I'm no longer seeing 'groups' listed in the sync status. Is this normal?

![image](https://github.com/goauthentik/authentik/assets/4944970/a76a7d42-cf7d-44c7-a716-ee18b277fe6c)

**Reply by AndrewBucklin** [reply, marked as answer]
I found the solution for this; one of the 4 servers in the list was offline (the 4th/last server). This is what prevented groups and group memberships from syncing.

I would consider this a bug; the whole reason to have multiple servers listed is for failover reasons.

**Reply by BeryJu** [reply]
Please open an issue for that, there have been some changes in regards to how authentik handles multiple LDAP servers but that is not intended
